### PR TITLE
flynt: init at 0.66

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Add [`flynt`](https://github.com/ikamensh/flynt) to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137177` at 47112f1a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/7aa96fb4b6c0d8521cae6ec8ddd454c8">python38Packages.flynt</a></li>
    <li><a href="https://gist.github.com/b386fa4e816ff35634f1dfc6e5c91f4f">python39Packages.flynt</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/python-modules/flynt/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+buildPythonPackage rec {
+  pname = "flynt";
+  version = "0.66";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "ikamensh";
+    repo = "flynt";
+    rev = "v${version}";
```
Comment (bobby285271) [inline pkgs/development/python-modules/flynt/default.nix:17]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/flynt/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, astor
+, pytestCheckHook
+}:
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flynt/default.nix:8]:
```suggestion
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/flynt/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    rev = version;
+    hash = "sha256-DV433wqLjF5k4g8J7rj5gZfaw+Y4/TDOoFKo3eKDjZ4=";
+  };
+
+  propagatedBuildInputs = [ astor ];
+
+  checkInputs = [ pytestCheckHook ];
+
+  meta = with lib; {
+    description = "flynt is a command line tool to automatically convert a project's Python code from old \"%-formatted\" and .format(...) strings into Python 3.6+'s \"f-strings\".";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flynt/default.nix:27]:
```suggestion
    description = "command line tool to automatically convert a project's Python code from old format style strings into Python 3.6+'s f-strings";
```

**Pull request merged**